packages/block-editor/src/components/block-preview/style.scss

```diff
@@ -1,34 +1,87 @@
-.block-editor-block-preview {
+// This is the preview that shows up to the right of the thumbnail when hovering.
+.block-editor-block-switcher__preview {
 	padding: $block-padding;
 	font-family: $editor-font;
 	overflow: hidden;
 	width: 100%;
+	pointer-events: none;
+	display: none;
+
+	@include break-medium {
+		display: block;
+	}
+
+	.block-editor-block-preview__content {
+		font-family: $editor-font;
+
+		> div {
+			font-family: $editor-font;
+		}
+
+		&:not(.is-tall-preview) {
+			// Vertical alignment.
+			margin-top: 50%;
+		}
+	}
+
+	.block-editor-block-preview__title {
+		margin-bottom: 10px;
+		color: $dark-gray-300;
+	}
+}
+
+// These rules ensure the preview scales smoothly regardless of the container size.
+.block-editor-block-preview__container {
+	// In the component, a top padding is provided as an inline style to provid an aspect-ratio.
+	// This positioning enables the content to sit on top of that padding to fit.
+	position: relative;
+
+	// The preview component measures the pixel width of this item, so as to calculate the scale factor.
+	// But without this baseline width, it collapses to 0.
+	width: 100%;
+}
+
+.block-editor-block-preview__content {
+	// This element receives inline styles for width, height, and transform-scale.
+	// Those inline styles are calculated to fit a perfect thumbnail.
+
+	// Position above the padding.
+	position: absolute;
+
+	// Vertical alignment. It works with the transform: translate(-50%, -50%)`,
+	top: 0;
+	left: 0;
+
+	// Important to set the origin.
+	transform-origin: top left;
+
+	// Resetting paddings, margins, and other.
```

Comment:
I'm not sure if it was introduced here, but while the preview shows well in most contexts, in the block transform menu, I see horizontal scroll:

![image](https://user-images.githubusercontent.com/1779930/69673568-90c9aa00-1068-11ea-8c42-4fbc17054054.png)

From some inspection, it seems to be because of the negative offsets from pseudo-elements surrounding the block in the block list:

https://github.com/WordPress/gutenberg/blob/be671eaaec240b51b92110f68c0f03bcafe3842a/packages/block-editor/src/components/block-list/style.scss#L1120-L1122

I'd think that ideally, we could have some concept of a "read-only" block list rendering, since these previews have no use for all of the extra control elements of a block.

But I comment here because it seems like you had to deal with this anyways. Maybe something we can be addressing with `overflow: hidden` ? 

(Aside: I find it helps to use the "Show Scroll Bars: Always" option in macOS, to help discover these sorts of issues: https://cloudup.com/cqF641QROEe)

Reply:
I don't have a good answer here, This was a very complex PR and I know that it can be tricky to get right on all situations.

> read-only mode or preview mode

I see this mentioned over and over and I think we'll probably need that for several things (previews and also some special FSE mode where some blocks are "enabled" while others are not)

> negative left/right

This looks specific to the hover and selected styles, so yeah if we can move this out of the flow somehow (overflow: hidden maybe but I'd be very cautious), it could work or just remove the selected/hover styles entirely in preview.

Reply:
👋 Honestly, I don't remember exactly the purposes of introducing these resets here. As Riad mentioned, it's a very complex PR and any modification can affect other instances where the preview is being used. AMost of these resets are usually an attempt to make cross-theme support. 

`read-only`, `view`, etc. have been talked as well. I thought that even we could remove all elements which are not needed just for the preview (appender, drop-zone, etc) in order to make the preview process more efficient.